packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -150,6 +151,14 @@ export const GridRootStyles = styled('div', {
         borderBottomColor: 'transparent',
       },
     },
+    [`&.${gridClasses.autosizing}`]: {
+      [`& .${gridClasses.columnHeaderTitleContainerContent} > *`]: {
+        overflow: 'visible !important',
+      },
+      [`& .${gridClasses.cell} > *`]: {
+        overflow: 'visible !important',
```

Comment:
Do we really want to use `!important` here
Wouldn't CSS specificity make it work anyways?